controllers/inferenceservice_controller.go

```diff
@@ -149,6 +130,61 @@ func (r *OpenshiftInferenceServiceReconciler) SetupWithManager(mgr ctrl.Manager)
 	return nil
 }
 
+func (r *OpenshiftInferenceServiceReconciler) getReconcileRequestsOnUpdateOfServingRuntime(o client.Object) []reconcile.Request {
+	inferenceServicesList := &kservev1beta1.InferenceServiceList{}
+	opts := []client.ListOption{client.InNamespace(o.GetNamespace())}
+
+	// Todo: Get only Inference Services that are deploying on the specific serving runtime
+	err := r.client.List(context.TODO(), inferenceServicesList, opts...)
+	if err != nil {
+		r.log.Info("Error getting list of inference services for namespace")
+		return []reconcile.Request{}
+	}
+
+	if len(inferenceServicesList.Items) == 0 {
+		r.log.Info("No InferenceServices found for Network Policy: " + o.GetName())
+		return []reconcile.Request{}
+	}
+
+	reconcileRequests := make([]reconcile.Request, 0, len(inferenceServicesList.Items))
+	for _, inferenceService := range inferenceServicesList.Items {
+		reconcileRequests = append(reconcileRequests, reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      inferenceService.Name,
+				Namespace: inferenceService.Namespace,
+			},
+		})
+	}
+	return reconcileRequests
+}
+
+func (r *OpenshiftInferenceServiceReconciler) getReconcileRequestsOnUpdateOfNetworkPolicy(o client.Object) []reconcile.Request {
```

Comment:
Do we really need this function? It seems to be equal, except 1 comment to the `getReconcileRequestsOnUpdateOfServingRuntime`

Reply:
for now they are doing same task to return list of all InferenceServices in that namespace but we have task in our backlog to update the logic of `getReconcileRequestsOnUpdateOfServingRuntime` function. 
https://issues.redhat.com/browse/RHOAIENG-1265